osu.Game/Screens/Edit/Editor.cs

```diff
@@ -307,8 +307,17 @@ protected void Save()
             // apply any set-level metadata changes.
             beatmapManager.Update(playableBeatmap.BeatmapInfo.BeatmapSet);
 
-            // save the loaded beatmap's data stream.
-            beatmapManager.Save(playableBeatmap.BeatmapInfo, editorBeatmap, editorBeatmap.BeatmapSkin);
+            try
+            {
+                // save the loaded beatmap's data stream.
+                beatmapManager.Save(playableBeatmap.BeatmapInfo, editorBeatmap, editorBeatmap.BeatmapSkin);
+            }
+            catch
+            {
+                // this is a temporary measure to avoid hard crashes on saving identical beatmaps.
+                // should be replaced in the future once we aren't relying on the isNewBeatmap flag.
+                beatmapManager.Delete(playableBeatmap.BeatmapInfo.BeatmapSet);
```

Comment:
This can still crash, if you try to save once, get the exception notifications, and try to save again - it'll die on an `OptimisticConcurrencyException`. I'm looking at what's causing that to get thrown right now.

Reply:
Honestly maybe better to just leave this one not fixed at this point. It's not like it's going to negatively affect any normal user.

Reply:
Are you suggesting to not merge this PR? Or to merge it and leave the followup issue pointed out above unfixed?

Reply:
Maybe best to just close this for now and deal with it in a better way when we can (via having a patcher/differ that supports the whole file, not just hitobjects).